utils/funcexp/func_floor.cpp

```diff
@@ -160,7 +160,7 @@ int64_t Func_floor::getIntVal(Row& row,
 
             if (op_ct.colWidth == datatypes::MAXDECIMALWIDTH)
             {
-                ret = datatypes::Decimal::getInt64FromWideDecimal(tmp.s128Value);
+                ret = static_cast<int64_t>(tmp.toTSInt128());
```

Comment:
I prefer this:

`return static_cast<int64_t>(tmp);`

Reasoning:
- We have the conversion operator defined already. Why not use it?
- There are no reasons to break the switch and do "return" immediately after the switch. Just return right here.

Reply:
This calls for an unrelated change in the else branch that I don't want to do.

Reply:
I would say that this behavior looks very strange we returning actual int128 value not looking at its scale.  That is why we casting tmp.toTSInt128() to int64 in the first place. I believe that this calls for additional investigation outside the scope of this issue.